### PR TITLE
ID-32: Proposal to adopt the OpenId based MOSIP's Identity provider open API

### PR DESCRIPTION
**Pull request by gsasikumar**

### CODE REVIEW

File: api/Identity-Provider.yaml [hunk 1]
```diff
@@ -0,0 +1,1561 @@
...
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  responseTime:
+                    type: string
+                    description: ate and time when the response is generated
```
Comment (rachellawson) [inline api/Identity-Provider.yaml:195]: Missing D on Date

File: api/Identity-Provider.yaml [hunk 2]
```diff
@@ -0,0 +1,1561 @@
...
+  /client-mgmt/oidc-client:
+    post:
+      summary: Create OIDC Client Endpoint
+      operationId: post-client
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+              description: OIDC client details
```
Comment (rachellawson) [inline api/Identity-Provider.yaml:48]: Some description attributes are appearing in quotes and some are not - is there an agreed patter to these we should follow?

Comment (rachellawson) [reply api/Identity-Provider.yaml:48]: Obviously, multi-line descriptions are using |- which is fine

**Comment by jdub13** [comment]
Opening that Pull Request for GovStack community review (people external to IDBB and to GovStack)

Dear GovStack community, this API aims to **Authenticate, get a consent and sharing attributes** from a generic Identity Building Block using standard protocol **OpenID Connect.**

Objective of that review is :
1/ To make sure the **expected functions are delivered**
2/ To make sure this is **100% compliant with openID Connect** protocol which means we could eventually switch with another IDP

The review is **opened for 2 weeks**, up to **Tuesday December 6th**.
Once this review will be achieved, this API will merged into IDBB and GitBook documentation will be produced.

**Pull request merged**